nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -174,7 +177,7 @@ let
 
         redirectListen = filter (x: !x.ssl) defaultListen;
 
-        acmeLocation = ''
+        acmeLocation = optionalString (vhost.enableACME || vhost.useACMEHost != null) ''
```

Comment:
What's the purpose of `acmeLocation` when `useACMEHost != null`?

Reply:
Hmm, I actually have no idea if it is needed. I probably just kept it in case Let’s Encrypt runs the challenge on all subdomains in the certificate.

Reply:
Necessary to allow challenges to be reachable for HTTP-01, not necessary when using `security.acme.certs.<cert>.dnsProvider`. See https://github.com/NixOS/nixpkgs/pull/245737